src/realloc-buffer.c

```diff
@@ -353,3 +354,41 @@ int realloc_buffer_write_maybe(ReallocBuffer *b, int fd) {
 
         return realloc_buffer_write(b, fd);
 }
+
+int realloc_buffer_printf(ReallocBuffer *b, const char *fmt, ...) {
+        va_list ap;
+        size_t m = 64;
+
+        if (!b)
+                return -EINVAL;
+        if (!fmt)
+                return -EINVAL;
+
+        for (;;) {
+                size_t mm;
+                void *p;
+                int n;
+
+                p = realloc_buffer_extend(b, m);
+                if (!p)
+                        return -ENOMEM;
+
+                va_start(ap, fmt);
+                n = vsnprintf(p, m, fmt, ap);
+                va_end(ap);
```

Comment:
Even with exponential increases, this seems rather wasteful...

Edit: OK, there are two users, and they both fit in 64 bytes, so this is not wasteful. But if additional users appear which routinely use more than 64 bytes, it'll probably make sense to extend the initial buffer size.